common/serial/port_io.h

```diff
@@ -0,0 +1,43 @@
+#ifndef PORT_IO__H
+#define PORT_IO__H
+
+#include <arch.h>
+
+class port_io {
+
+public:
+	port_io(arch_handler *arch) : m_arch(arch) 
+	{
+
+	}
+
+	~port_io(void) 
+	{
+		delete m_arch;
+	}
+
+	void port_write(uint16_t port, uint8_t value)
+	{
+		m_arch->port_write(port, value);
```

Comment:
NASM vs GCC parameterized inline assembly will likely result in the same number of lines. The system V call convention is dirt simple, and you only use what you need. No stack crap, etc.... So the code would be something like outb RDI, RSI, ret 0